Doc: wildcards in xpack.management.pipeline.id (#12396)
elastic/logstash: docs/static/settings/configuration-wildcard-pipeline-id.asciidoc

[role="xpack"]
[[wildcard-in-pipeline-id]]
==== Wildcard support in pipeline ID
++++
<titleabbrev>Wildcard support in pipeline ID</titleabbrev>
++++

> Pipeline ID must begin with a letter or underscore and contain only letters, underscores, dashes, and numbers.

Reviewer: ```suggestion
Pipeline IDs must begin with a letter or underscore and contain only letters, underscores, dashes, and numbers.
```

> You can use `*` in `xpack.management.pipeline.id` to match any number of letter, underscore, dash, and number.

Reviewer: ```suggestion
You can use `*` in `xpack.management.pipeline.id` to match any number of letters, underscores, dashes, and numbers.
```

[source,shell]
-----
xpack.management.pipeline.id: ["*logs", "*apache*", "tomcat_log"]
-----

> In this example, `"*logs"` matches all IDs end in `logs`. `"*apache *"` matches any IDs with `apache` in the name.

Author: how to escape `*` in asciidoc?  `"*apache*"` bold apache instead of showing two `*` in text.

Reviewer: It appears to be rendering correctly.  You must have figured this out.

> In this example, `"*logs"` matches all IDs end in `logs`. `"*apache *"` matches any IDs with `apache` in the name.

Reviewer: ```suggestion
In this example, `"*logs"` matches all IDs ending in `logs`. `"*apache*"` matches any IDs with `apache` in the name.
```

Author: For the space in `"*apache *"`, if I remove the space, it renders a bold "**apache**" without `*`.
That why I ask the question of how to escape `*`

Author: :) I found a way to escape it.

> Wildcard is available starting with Logstash 7.11 alongside Elasticsearch 7.10. Logstash can pick up new pipeline without restart Logstash if the new pipeline ID matches the wildcard pattern.

Reviewer: We try to avoid listing version numbers in documentation.  Documentation is available (using the version dropdown) for all versions, so this info is generally not needed.  For example, this docs PR would be merged into `master` and `7.x`, but no earlier version so we don't have to say it starts at Logstash 7.11 because it won't be in docs for Logstash 7.10 and earlier.  

Sometimes we make exceptions based on the situation. Let's discuss if you'd like. :-)

Author: totally agree. Should I mention Elasticsearch version? Do we expect users to have the same version in general, so we don't mention?

> Wildcard in pipeline IDs is available starting with Elasticsearch 7.10. Logstash can pick up new pipeline without restart Logstash if the new pipeline ID matches the wildcard pattern.

Reviewer: ```suggestion
Wildcard in pipeline IDs is available starting with Elasticsearch 7.10. Logstash can pick up new pipeline without a restart if the new pipeline ID matches the wildcard pattern.
```
